pkg/virt-operator/util/config.go

```diff
@@ -84,8 +84,8 @@ const (
 // DefaultMonitorNamespaces holds a set of well known prometheus-operator namespaces.
 // Ordering in the list matters. First entries have precedence.
 var DefaultMonitorNamespaces = []string{
-	"openshift-monitoring", // default namespace in openshift
-	"monitoring",           // default namespace of https://github.com/prometheus-operator/kube-prometheus
+	"openshift-cnv", // default namespace in openshift
```

Comment:
@Barakmor1 we just use the `openshift-monitoring` and the `monitoring` namespaces just to look for common places of prometheus-operator. We always want to install in the kubevirt install namespace (not specifically `openshift-cnv` or `kubevirt`).

So here `openshift-monitoring` and `monitoring` need to stay as they are in this detection map.

Reply:
> I think there is a confusion regarding to why we care about the `openshift-monitoring` namespace. We just look for common install locations on common k8s distributions.
> 
> In general we should stop looking for that and instead do an autodiscovery for the `ServiceMonitor` CR and not try to indirectly guess if the prometheus-operator exists.



> @Barakmor1 we just use the `openshift-monitoring` and the `monitoring` namespaces just to look for common places of prometheus-operator. We always want to install in the kubevirt install namespace (not specifically `openshift-cnv` or `kubevirt`).
> 
> So here `openshift-monitoring` and `monitoring` need to stay as they are in this detection map.



Let me give some useful information about this discussion:

> The Prometheus resource includes a field called serviceMonitorSelector, which defines a selection of ServiceMonitors to be used. By default and before the version v0.19.0, ServiceMonitors must be installed in the same namespace as the Prometheus instance. With the Prometheus Operator v0.19.0 and above, ServiceMonitors can be selected outside the Prometheus namespace via the serviceMonitorNamespaceSelector field of the Prometheus resource.

See https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md

In OpenShift, the configuration is as below by default:
```
$ oc get prometheus k8s -n  openshift-monitoring -o yaml
...
  serviceMonitorNamespaceSelector:
    matchLabels:
      openshift.io/cluster-monitoring: "true"
...
```

And we have `openshift.io/cluster-monitoring=true` label on openshift-cnv namespace. That is why we can create our ServiceMonitor object in openshift-cnv namespace.

I could not find an easy way to check whether prometheus-operator will discover ServiceMonitor objects in Kubervirt's namespace or not without following this long path. We can open a discussion in prometheus-operator side about it.

Reply:
Also note that we need to keep this page up-to-date according to results of this PR https://github.com/kubevirt/user-guide/blob/master/docs/operations/component_monitoring.md#integrating-with-the-prometheus-operator

Reply:
What I tried to say: If you change the `DefaultMonitoringNamespaces` value to `openshift-cnv` from `openshift-monitoring` the code will not detect the presence of the prometheus-opertor properly and will not do anything.

Reply:
@rmohr  is right. We expect having the namespace and having prometheus-k8s service account in that namespace by default  here https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-operator/resource/generate/install/strategy.go#L258

This change will not work since `prometheus-k8s` serviceaccount doesn't exists in kubevirt's namespace (openshift-cnv for openshift)

As I said, if want to prioritize kubevirt's namespace, we have to ensure that it will be discovered by prometheus by checking the things I shared above. 

Also note that, Kubevirt has some fields to configure the namespace and service account. For the  OpenShift+HCO  scenario, HCO can set them accordingly to move the servicemonitor object into openshift-cnv namespace.

Reply:
@rmohr @erkanerol 
Thanks for the clarifications,
So if i understand correctly a better solution would be to place the serviceMonitor objects in one of our namespaces(not openshift-cnv ) to decouple ourselves from the downstream, and to do that we need to add `openshift.io/cluster-monitoring=true` label and `prometheus-k8s` serviceAccount to that namespace and consider the update to make the 
old serviceMonitor objects visible to virt-controller  .
Did i get it right?

Reply:
@rmohr @erkanerol
Thanks for the clarifications,
So if i understand correctly the reason we refer to a downstream namespace is because ,
the downstream should define `prometheus-k8s` serviceAccount in there namespace right?

Reply:
Yes, all this is just to detect the presence of prometheus-operator.
However, if we are going to improve things in that area I would suggest the following:

 * Of course creating our monitors and so on in the install namespace of kubevirt (not the rometheus-operator namespace) and just adding the `openshift.io/cluster-monitoring=true` label to the objects (just always).
 * Detecting if ServiceMonitor objects are registered
 * Dropping the old detection logic
 
 Here ow one can find out if some resources are registered in the cluster: https://github.com/openshift/hypershift/blob/main/support/capabilities/management_cluster_capabilities.go#L42

Reply:
So if i understand correctly it's possible to make the serviceMonitor objects visible to prometheus-operator in ANY
 namespace if we register ServiceMonitor objects right?

Reply:
No. It depends on `serviceMonitorNamespaceSelector` field of Prometheus object as I explained here: https://github.com/kubevirt/kubevirt/pull/6662#discussion_r748075402

Since we cannot change it for our product, there may be some cases that we cannot use kubevirt's namespace for servicemonitor objects even it has `openshift.io/cluster-monitoring=true` label.